pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -135,14 +136,38 @@ func (a *index) constructService(svc *v1.Service, w *Waypoint) *workloadapi.Serv
 		waypointAddress = a.getWaypointAddress(w)
 	}
 
+	var lb *workloadapi.LoadBalancing
+	if svc.Labels[constants.ManagedGatewayLabel] == constants.ManagedGatewayMeshControllerLabel {
+		// This is waypoint. Enable locality routing
+		lb = &workloadapi.LoadBalancing{
+			// Prefer endpoints in close zones, but allow spilling over to further endpoints where required.
+			RoutingPreference: []workloadapi.LoadBalancing_Scope{
+				workloadapi.LoadBalancing_NETWORK,
+				workloadapi.LoadBalancing_REGION,
+				workloadapi.LoadBalancing_ZONE,
+				workloadapi.LoadBalancing_SUBZONE,
+			},
+			Mode: workloadapi.LoadBalancing_FAILOVER,
```

Comment:
Is this good to always enable locality lb for the waypoint? I can think of cases where this is not the most optimized path

Reply:
Currently we don't have an API that we want to expose for ambient so was wanting to default here. In the near future we can use the Kubernetes API